pkg/visor/visorconfig/v1.go

```diff
@@ -151,32 +160,108 @@ func (v1 *V1) UpdateAppArg(launch *launcher.Launcher, appName, argName, value st
 	return v1.flush(v1)
 }
 
-func updateArg(conf *V1Launcher, appName, argName, value string) bool {
+// updateStringArg updates the cli non-boolean flag of the specified app config and also within the launcher.
+// It removes argName from app args if value is an empty string.
+// The updated config gets flushed to file if there are any changes.
+func updateStringArg(conf *V1Launcher, appName, argName, value string) bool {
 	configChanged := false
 
 	for i := range conf.Apps {
-		if conf.Apps[i].Name == appName {
-			configChanged = true
-
-			argChanged := false
-			l := len(conf.Apps[i].Args)
-			for j := 0; j < l; j++ {
-				if conf.Apps[i].Args[j] == argName && j+1 < len(conf.Apps[i].Args) {
-					if value == "" {
-						conf.Apps[i].Args = append(conf.Apps[i].Args[:j], conf.Apps[i].Args[j+2:]...)
-						j--
+		if conf.Apps[i].Name != appName {
+			continue
+		}
+
+		configChanged = true
+
+		argChanged := false
+		l := len(conf.Apps[i].Args)
+		for j := 0; j < l; j++ {
+			equalArgName := conf.Apps[i].Args[j] == argName && j+1 < len(conf.Apps[i].Args)
+			if !equalArgName {
+				continue
+			}
+
+			if value == "" {
+				conf.Apps[i].Args = append(conf.Apps[i].Args[:j], conf.Apps[i].Args[j+2:]...)
+				j-- //nolint:ineffassign
+			} else {
+				conf.Apps[i].Args[j+1] = value
+			}
+
+			argChanged = true
+			break
+		}
+
+		if !argChanged {
+			conf.Apps[i].Args = append(conf.Apps[i].Args, argName, value)
+		}
+
+		break
+	}
+
+	return configChanged
+}
+
+// updateBoolArg updates the cli boolean flag of the specified app config and also within the launcher.
+// All flag names and values are formatted as "name=value" to allow arbitrary values with respect to different
+// possible default values.
+// The updated config gets flushed to file if there are any changes.
+func updateBoolArg(conf *V1Launcher, appName, argName string, value bool) bool {
+	const argFmt = "%s=%v"
+
+	configChanged := false
+
+	for i := range conf.Apps {
+		if conf.Apps[i].Name != appName {
+			continue
+		}
+
+		// we format it to have two dashes, just to unify representation
+		fmtedArgName := argName
```

Comment:
Would it not be cleaner to stick with single dashes? Maybe it does not look as nice, but works just as well.

Reply:
It doesn't really matter at all, so will do